ocaml/xapi/import_raw_vdi.ml

```diff
@@ -114,13 +114,24 @@ let import vdi (req: Request.t) (s: Unix.file_descr) _ =
               then localhost_handler rpc session_id vdi req s
               else
                 let host = Importexport.find_host_for_sr ~__context sr in
-                let address = Db.Host.get_address ~__context ~self:host in
-                return_302_redirect req s address;
-                None
+                if host <> Ref.null then
+                  let address = Db.Host.get_address ~__context ~self:host in
+                  return_302_redirect req s address;
+                else begin
+                  Xapi_http.with_context "VDI.import" req s
+                    (fun __context ->
+                      TaskHelper.failed ~__context Api_errors.(Server_error(no_hosts_available,[]));
+                      Http_svr.headers s (Http.http_400_badrequest ());
+                    )
+                end;
```

Comment:
I think it's easier if we handle this exception in a try/with block around this function. Then we don't have to modify `find_host_for_sr`, we can just catch the exception it throws (`Api_errors.no_hosts_available`). Also, I think that we should catch any exception that occurs here, and set the task to failed, and return an HTTP error code.